File: model/src/main/java/io/sparkled/model/util/SequenceUtils.java
package io.sparkled.model.util;

import io.sparkled.model.entity.Sequence;
import io.sparkled.model.entity.Song;

import static io.sparkled.model.constant.ModelConstants.MS_PER_SECOND;

/**
 * Helper functions for sequences.
 */
public class SequenceUtils {

    private SequenceUtils() {
    }

    /**
     *
     * @param song The song used by the sequence, which contains the duration in milliseconds.
     * @param sequence The sequence, which contains the FPS.
     * @return The number of frames available in the sequence.
     */
    public static int getFrameCount(Song song, Sequence sequence) {
        return (song.getDurationMs() / MS_PER_SECOND) * sequence.getFramesPerSecond();
    }
}

File: model/src/main/kotlin/io/sparkled/model/util/SequenceUtils.kt
package io.sparkled.model.util

import io.sparkled.model.SongModel
import io.sparkled.model.constant.ModelConstants.DEFAULT_FRAMES_PER_SECOND
import io.sparkled.model.constant.ModelConstants.MS_PER_SECOND

/**
 * Helper functions for sequences.
 */
object SequenceUtils {

    /**
     * @param song The song used by the sequence, which contains the duration in milliseconds.
     * @return The number of frames available in the sequence.
     */
    fun getFrameCount(song: SongModel): Int {
        return (song.durationMs / MS_PER_SECOND.toFloat() * DEFAULT_FRAMES_PER_SECOND).toInt()
    }
}
